src/components/ReportActionItem/TaskPreview.tsx

```diff
@@ -24,141 +21,134 @@ import * as LocalePhoneNumber from '@libs/LocalePhoneNumber';
 import * as Session from '@userActions/Session';
 import * as Task from '@userActions/Task';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
+import type {Policy, Report, ReportAction} from '@src/types/onyx';
+import {isNotEmptyObject} from '@src/types/utils/EmptyObject';
 
-const propTypes = {
-    /** The ID of the associated taskReport */
-    taskReportID: PropTypes.string.isRequired,
-
-    /** Whether the task preview is hovered so we can modify its style */
-    isHovered: PropTypes.bool,
-
-    /** The linked reportAction */
-    action: PropTypes.shape(reportActionPropTypes).isRequired,
+type PolicyRole = {
+    role: string;
+};
 
+type TaskPreviewOnyxProps = {
     /* Onyx Props */
 
-    taskReport: PropTypes.shape({
-        /** Title of the task */
-        reportName: PropTypes.string,
-
-        /** AccountID of the manager in this iou report */
-        managerID: PropTypes.number,
-
-        /** AccountID of the creator of this iou report */
-        ownerAccountID: PropTypes.number,
-    }),
+    taskReport: OnyxEntry<Report>;
 
     /** The policy of root parent report */
-    rootParentReportpolicy: PropTypes.shape({
-        /** The role of current user */
-        role: PropTypes.string,
-    }),
-
-    /** The chat report associated with taskReport */
-    chatReportID: PropTypes.string.isRequired,
-
-    /** Popover context menu anchor, used for showing context menu */
-    contextMenuAnchor: refPropTypes,
-
-    /** Callback for updating context menu active state, used for showing context menu */
-    checkIfContextMenuActive: PropTypes.func,
-
-    /* Onyx Props */
-    ...withLocalizePropTypes,
-
-    ...withCurrentUserPersonalDetailsPropTypes,
-};
-
-const defaultProps = {
-    ...withCurrentUserPersonalDetailsDefaultProps,
-    taskReport: {},
-    rootParentReportpolicy: {},
-    isHovered: false,
+    rootParentReportpolicy: OnyxEntry<PolicyRole>;
 };
 
-function TaskPreview(props) {
+type TaskPreviewProps = WithCurrentUserPersonalDetailsProps &
+    TaskPreviewOnyxProps & {
+        /** The ID of the associated policy */
+        // eslint-disable-next-line
+        policyID: string;
+        /** The ID of the associated taskReport */
+        taskReportID: string;
+
+        /** Whether the task preview is hovered so we can modify its style */
+        isHovered: boolean;
+
+        /** The linked reportAction */
+        action: OnyxEntry<ReportAction>;
+
+        /** The chat report associated with taskReport */
+        chatReportID: string;
+
+        /** Popover context menu anchor, used for showing context menu */
+        contextMenuAnchor: Element;
+
+        /** Callback for updating context menu active state, used for showing context menu */
+        checkIfContextMenuActive: () => void;
+    };
+
+function TaskPreview({
+    taskReport,
+    taskReportID,
+    action,
```

Comment:
@blazejkustra Is there a general rule about when to use destructuring with defaults and when not to? It feels to me like it would make the code easier to follow if we used defaults in the destructuring. The only case I think it won't work is when you need to know if the value was `undefined` or `null`, but a simple fasly check like `action ?? {}` should be fine, right?

Reply:
so may I update the code to use default value in destructuring if possible?

Reply:
The rule of thumb is to avoid unnecessary default values such as undefined, null, {}, [], '' and () => {} - when we started migration we left some of these, but we can clean them as a follow up once whole TS migration is finished.

Two examples for useful default values:
```ts
function Avatar({
    size = CONST.AVATAR_SIZE.DEFAULT,
    fallbackIcon = Expensicons.FallbackAvatar,
    type = CONST.ICON_TYPE_AVATAR,
}: AvatarProps) {}

function GenericPressable(
    {
        shouldUseHapticsOnLongPress = false,
        shouldUseHapticsOnPress = false,
        shouldUseAutoHitSlop = false,
        enableInScreenReaderStates = CONST.SCREEN_READER_STATES.ALL,
        accessible = true,
        ...rest
    }: PressableProps) {}
```